src/components/GameLibrary/GameLibrarySecondary.java

```diff
@@ -0,0 +1,112 @@
+package components.GameLibrary;
+
+/**
+ * Implements all 3 secondary methods as well as toString and equals.
+ */
+public abstract class GameLibrarySecondary implements GameLibrary {
+
+    /**
+     * Selects an arbitrary game from the library.
+     *
+     * @return a random game from the library
+     * @requires this.size() > 0
+     * @ensures the selected game is added back to the library
+     *
+     */
```

Comment:
The contracts aren't needed since they're inherited. You can also mark these methods with `@Override`